File: aoc/2025/day01/day01_B.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
  string direction;
  int current_position = 50;
  int ans = 0;
  while (cin >> direction) {
    int position;
    cin >> position;
    if (direction == "R") {
      current_position += position;
      ans += current_position / 100;
      current_position %= 100;
    } else {
      if (current_position == 0)
        current_position = 100;
      current_position -= position;
      if (current_position < 0)
        ans += (abs(current_position) + 100 - 1) / 100;
      current_position %= 100;
      if (current_position < 0)
        current_position += 100;
      if (current_position == 0)
        ans++;
    }
  }
  cout << ans << '\n';
  return 0;
}

File: aoc/2025/day01/day01_B.ml
let rec solve current_position ans =
  match read_line () with
  | exception End_of_file -> ans
  | line ->
      let dir, pos = Scanf.sscanf line "%s %d" (fun d p -> d, p) in
      let new_pos, addition =
        let p =
          match dir, current_position with
            | "R", _ -> current_position + pos
            | "L", 0 -> 100 - pos
            | "L", _ -> current_position - pos
            | _, _ -> current_position
        in
        let addition =
          match dir with
            | "R" -> p / 100
            | "L" -> if p < 0 then ((abs p) + 100 - 1)/100 else 0
            | _ -> 0
        in
        let p = p mod 100 in
        match dir, p with
          | "R", _ -> (p, addition)
          | "L", 0 -> (p, addition + 1)
          | "L", _ -> ((p + 100) mod 100, addition)
          | _, _ ->  (p, addition)
      in
      solve new_pos (ans + addition)

let () =
  let result = solve 50 0 in
  Printf.printf "%d\n" result
